workspaces/ui-v2/src/optic-components/pages/diffs/ReviewEndpointDiffPage/ReviewEndpointDiffPage.tsx

```diff
@@ -71,29 +74,66 @@ export const ReviewEndpointDiffPage: FC<ReviewEndpointDiffPageProps> = ({
   });
   const endpointId = getEndpointId({ method, pathId });
 
-  const allDiffsHandled = shapeDiffs.every(
-    (shapeDiff) =>
-      shapeDiff?.diffDescription?.diffHash &&
-      isDiffHandled(shapeDiff.diffDescription.diffHash)
-  );
-  const [currentIndex, setCurrentIndex] = useState(0);
+  const getNextIncompleteDiff = (recentlyCompletedDiff?: string): number => {
```

Comment:
So this is slightly less ideal, when we trigger this function, the xState store hasn't updated yet so the subsequent `isDiffHandled` call returns false (i think it uses useState under the hood) - this is likely a problem with any state management solutions. The other thing we could try do a `setTimeout(setCurrentIndex, 0)` which should defer this call until xState updates, but this is hacky.